packages/backend/src/tokens/request.ts

```diff
@@ -574,6 +662,106 @@ export const debugRequestState = (params: RequestState) => {
 };
 
+/**
+ * Determines if the given URL and settings indicate a desire to activate a specific
+ * organization or personal workspace.
+ */
+export function getOrganizationSyncTarget(
+  url: URL,
+  options: OrganizationSyncOptions | undefined,
+): OrganizationSyncTarget | null {
+  if (!options) {
+    return null;
+  }
+
+  // Check for personal workspace activation
+  if (options.personalWorkspacePatterns) {
+    let matcher: MatchFunction<Partial<Record<string, string | string[]>>>;
+    try {
+      matcher = match(options.personalWorkspacePatterns);
+    } catch (e) {
+      // Likely to be encountered during development, so throwing the error is more prudent than logging
+      throw new Error(`Invalid personal workspace pattern "${options.personalWorkspacePatterns}": "${e}"`);
+    }
+
+    let personalResult: Match<Partial<Record<string, string | string[]>>>;
+    try {
+      personalResult = matcher(url.pathname);
+    } catch (e) {
+      // Intentionally not logging the path to avoid potentially leaking anything sensitive
+      console.error(`Failed to apply personal workspace pattern "${options.personalWorkspacePatterns}" to a path`, e);
+      return null;
+    }
+
+    if (personalResult) {
+      return { type: 'personalWorkspace' };
+    }
+  }
+
+  // Check for organization activation
+  if (options.organizationPatterns) {
+    let matcher: MatchFunction<Partial<Record<string, string | string[]>>>;
+    try {
+      matcher = match(options.organizationPatterns);
+    } catch (e) {
+      // Likely to be encountered during development, so throwing the error is more prudent than logging
+      throw new Error(`Clerk: Invalid organization pattern "${options.organizationPatterns}": "${e}"`);
+    }
+
+    let orgResult: Match<Partial<Record<string, string | string[]>>>;
+    try {
+      orgResult = matcher(url.pathname);
+    } catch (e) {
+      // Intentionally not logging the path to avoid potentially leaking anything sensitive
+      console.error(`Clerk: Failed to apply organization pattern "${options.organizationPatterns}" to a path`, e);
+      return null;
+    }
+
+    if (orgResult && 'params' in orgResult) {
+      const params = orgResult.params;
+
+      if ('id' in params && typeof params.id === 'string') {
+        return { type: 'organization', organizationId: params.id };
+      }
+      if ('slug' in params && typeof params.slug === 'string') {
+        return { type: 'organization', organizationSlug: params.slug };
+      }
+      console.warn(
+        'Clerk: Detected an organization pattern match, but no organization ID or slug was found in the URL. Does the pattern include `:id` or `:slug`?',
+      );
+    }
+  }
+  return null;
+}
+
+/**
+ * Represents an organization or a personal workspace - e.g. an
+ * entity that can be activated by the handshake API.
+ */
+export type OrganizationSyncTarget =
+  | { type: 'personalWorkspace' }
```

Comment:
@izaaklauer I remember us labelling this as "Personal Account". Can you confirm with @colinclerk on this ?

Reply:
👍 Internal conversation around this here: https://clerkinc.slack.com/archives/C078JFZLLQ3/p1727818679671599

Reply:
tl;dr: Agreed that account is canonical, and i'll update the references accordingly. Thank you!

Reply:
Fixed in 5858a24cb1215314f3adf73bac1f8e397d1aeefc